bots/csr/src/main/java/org/openjdk/skara/bots/csr/CSRBot.java

```diff
@@ -79,69 +80,66 @@ private String describe(PullRequest pr) {
-                        }
-                        continue;
-                    }
-                    var name = resolution.get("name");
-                    if (name == null || name.isNull()) {
-                        if (!pr.labelNames().contains(CSR_LABEL)) {
-                            log.info("CSR issue resolution name is null for " + describe(pr) + ", adding the CSR label");
-                            pr.addLabel(CSR_LABEL);
-                        } else {
-                            log.info("CSR issue resolution name is null for " + describe(pr) + ", not removing the CSR label");
-                        }
-                        continue;
-                    }
-
-                    if (csr.state() != Issue.State.CLOSED) {
-                        if (!pr.labelNames().contains(CSR_LABEL)) {
-                            log.info("CSR issue state is not closed for " + describe(pr) + ", adding the CSR label");
-                            pr.addLabel(CSR_LABEL);
-                        } else {
-                            log.info("CSR issue state is not closed for " + describe(pr) + ", not removing the CSR label");
-                        }
-                        continue;
-                    }
-
-                    if (!name.asString().equals("Approved")) {
-                        if (name.asString().equals("Withdrawn")) {
-                            // This condition is necessary to prevent the bot from adding the CSR label again.
-                            // And the bot can't remove the CSR label automatically here.
-                            // Because the PR author with the role of Committer may withdraw a CSR that
-                            // a Reviewer had requested and integrate it without satisfying that requirement.
-                            log.info("CSR closed and withdrawn for " + describe(pr) + ", not revising (not adding and not removing) CSR label");
-                        } else if (!pr.labelNames().contains(CSR_LABEL)) {
-                            log.info("CSR issue resolution is not 'Approved' for " + describe(pr) + ", adding the CSR label");
-                            pr.addLabel(CSR_LABEL);
-                        } else {
-                            log.info("CSR issue resolution is not 'Approved' for " + describe(pr) + ", not removing the CSR label");
-                        }
-                        continue;
-                    }
-
-                    if (pr.labelNames().contains(CSR_LABEL)) {
-                        log.info("CSR closed and approved for " + describe(pr) + ", removing CSR label");
-                        pr.removeLabel(CSR_LABEL);
-                    }
+            var csrOptional = jbsIssue.get().links().stream()
+                    .filter(link -> link.relationship().isPresent() && "csr for".equals(link.relationship().get()))
+                    .findAny().flatMap(Link::issue);
```

Comment:
I realize you'd be breaking abstractions by using csrLink() from CSRCommand, but maybe you can create a duplicated csrLink method here? I think it would help with readability, at least. And if we ever refactor this more, it would be a clear candidate to move to a class that is accessible from both here and CSRCommand.

Reply:
Encapsulating such a method can be reused in the future. Fixed